src/Query/Builder.php

```diff
@@ -1247,6 +1273,19 @@ public function options(array $options)
         return $this;
     }
 
+    /**
+     * Set session for the transaction
+     * @param $session
+     * @return mixed
+     */
+    protected function setSession($options = [])
```

Comment:
I noticed that most all of the calls to this method include the following comment, which is not very clear:

```
// if transaction in session
```

I propose we delete all of those comments and rename this method to more accurately convey what it does.

```suggestion
    /**
     * Apply the connection's session to options if it's not already specified.
     */
    private function inheritConnectionOptions(array $options = []): array
```

Since this PR introduces the method and I don't see any calls to it from an inheriting class, I don't think we need to lead with`protected` in this PR. The visibility could always be relaxed down the line. But if `protected` is more idiomatic for Laravel, then we should stick with that. I suppose we can use `private` now and see if this draws attention in the final PR.

Lastly, I assume the type hints are not controversial. Seems like a good thing to add on any new APIs.

Reply:
Done